web_poet/page_inputs/http.py

```diff
@@ -18,12 +19,46 @@
 _AnyStrDict = Dict[AnyStr, Union[AnyStr, List[AnyStr], Tuple[AnyStr, ...]]]
 
 
-class ResponseURL(str):
+class _Url:
+    def __init__(self, url: Union[str, yarl.URL]):
+        self._url = yarl.URL(str(url))
+
+    def __str__(self) -> str:
+        return str(self._url)
+
+    def __repr__(self) -> str:
+        return str(self._url)
+
+    def __eq__(self, other) -> bool:
+        return str(self._url) == str(other)
```

Comment:
We should probably make sure we implement this in line with our API expectations.

Mind the following:

```python
>>> example_url_1 = URL("https://example.com")
>>> example_url_2 = URL("https://example.com/")
>>> example_url_1 == example_url_2
True
>>> str(example_url_1) == str(example_url_2)
False
```

Reply:
While that is true for `https://example.com/foo` and `https://example.com/foo/`, the case of ``https://example.com` and `https://example.com/` is special, since the path of a URL, when not specified, is `/`.

```python
>>> URL('https://example.com').path
'/'
>>> URL('https://example.com/').path
'/'
>>> URL('https://example.com/foo').path
'/foo'
>>> URL('https://example.com/foo/').path
'/foo/'
```

Reply:
Ahhh thanks for making the distinction clear! Handled this on https://github.com/scrapinghub/web-poet/pull/45/commits/292a3b4b9e15041bc0a7a39ec06e2f8f20c29132.